common/src/importers/keeperImporters/keeperJsonImporter.ts

```diff
@@ -0,0 +1,68 @@
+import { BaseImporter } from "../baseImporter";
+import { Importer } from "../importer";
+
+import { ImportResult } from "../../models/domain/importResult";
+
+import { KeeperJsonExport, RecordsEntity } from "./types/keeperJsonTypes";
+
+export class KeeperJsonImporter extends BaseImporter implements Importer {
+  parse(data: string): Promise<ImportResult> {
+    const result = new ImportResult();
+    const keeperExport: KeeperJsonExport = JSON.parse(data);
+    if (keeperExport == null || keeperExport.records == null || keeperExport.records.length === 0) {
+      result.success = false;
+      return Promise.resolve(result);
+    }
+
+    keeperExport.records.forEach((record) => {
+      this.parseFolders(result, record);
+
+      const cipher = this.initLoginCipher();
+      cipher.name = record.title;
+      cipher.login.username = record.login;
+      cipher.login.password = record.password;
+
+      cipher.login.uris = this.makeUriArray(record.login_url);
+      cipher.notes = record.notes;
+
+      let customfieldKeys = Object.keys(record.custom_fields);
```

Comment:
Per the docs I can't tell if the property will be left off when empty. I'll try and get a new export without custom fields and see how Keeper exports it. If it can I'll also adjust the types. Good catch!

Reply:
I got a new export fresh from Keeper and you were right, some fields were optional. Adjusted the types, added a test case and the fix for missing custom fields.